### PR TITLE
Fix pNext definition

### PR DESCRIPTION
**Pull request by jandres742**
Resolves: #125

### CODE REVIEW

**Comment by jandres742** [comment]
@wdamon-intel : Please review.

File: scripts/tools/common.yml [hunk 1]
```diff
@@ -121,7 +121,7 @@ members:
       desc: "[in] type of this structure"
     - type: "void*"
       name: pNext
-      desc: "[in,out][optional] pointer to extension-specific structure"
+      desc: "[in,out][optional] must be null or a pointer to an extension-specific structure (i.e. contains sType and pNext)."
```
Comment (bmyates) [inline scripts/tools/common.yml:124]: `stype` is all lower case

Comment (jandres742) [reply scripts/tools/common.yml]: @bmyates thanks, fixed.

**Review by bmyates — APPROVED** [review]

**Comment by matcabral** [comment]
@jandres742 , would it make sense to also address  https://github.com/oneapi-src/level-zero-spec/issues/131 here? since its also related to pNext

File: scripts/core/common.yml [hunk 2]
```diff
@@ -552,7 +552,7 @@ members:
       desc: "[in] type of this structure"
     - type: "void*"
```
Comment (wdamon-intel) [inline scripts/core/common.yml:553]: Should this be `const void*`?

Comment (jandres742) [reply scripts/core/common.yml:553]: @wdamon-intel : possibly, but with that, we are actually changing the type of this member with a new modifier, so we couldn't just change it w/o breaking compatibility. I would say let's just leave it as is, unless there's a strong justification.

**Comment by jandres742** [comment]
> @jandres742 , would it make sense to also address #131 here? since its also related to pNext

thanks @matcabral . Added for most of the issues reported in latest commit.

File: scripts/sysman/PROG.rst [hunk 3]
```diff
@@ -382,7 +382,8 @@ supported:
                allocate_memory(numFreqDomains * sizeof(${s}_freq_handle_t))
            if (${s}DeviceEnumFrequencyDomains(hSysmanDevice, &numFreqDomains, pFreqHandles) == ${X}_RESULT_SUCCESS)
                for (index = 0 .. numFreqDomains-1)
-                   ${s}_freq_properties_t props
+                   ${s}_freq_properties_t props {};
+                   props.stype = ${S}_STRUCTURE_TYPE_FREQ_PROPERTIES;
```
Comment (matcabral) [inline scripts/sysman/PROG.rst:386]: missing nullptr

File: scripts/sysman/PROG.rst [hunk 4]
```diff
@@ -424,7 +425,8 @@ sub-device (notice the additional sub-device check):
...
```
Comment (matcabral) [inline scripts/sysman/PROG.rst:429]: missing nullptr

File: scripts/sysman/PROG.rst [hunk 5]
```diff
@@ -597,7 +599,8 @@ The pseudo code below shows how to output the PCI BDF address:
 .. parsed-literal::
 
    function ShowPciInfo(${s}_device_handle_t hSysmanDevice)
-       ${s}_pci_properties_t pciProps;
+       ${s}_pci_properties_t pciProps {};
+       pciProps.stype = ${S}_STRUCTURE_TYPE_PCI_PROPERTIES;
```
Comment (matcabral) [inline scripts/sysman/PROG.rst:603]: missing setting pNext to nullptr

File: scripts/sysman/PROG.rst [hunk 6]
```diff
@@ -792,7 +795,8 @@ domain on a device:
                allocate_memory(numPowerDomains * sizeof(${s}_pwr_handle_t))
            if (${s}DeviceEnumPowerDomains(hSysmanDevice, &numPowerDomains, phPower) == ${X}_RESULT_SUCCESS)
                for (pwrIndex = 0 .. numPowerDomains-1)
-                   ${s}_power_properties_t props
+                   ${s}_power_properties_t props {};
+                   props.stype = ${S}_STRUCTURE_TYPE_POWER_PROPERTIES;
```
Comment (matcabral) [inline scripts/sysman/PROG.rst:799]: missing setting pNext to nullptr

File: scripts/sysman/PROG.rst [hunk 7]
```diff
@@ -382,7 +382,8 @@ supported:
                allocate_memory(numFreqDomains * sizeof(${s}_freq_handle_t))
            if (${s}DeviceEnumFrequencyDomains(hSysmanDevice, &numFreqDomains, pFreqHandles) == ${X}_RESULT_SUCCESS)
                for (index = 0 .. numFreqDomains-1)
-                   ${s}_freq_properties_t props
+                   ${s}_freq_properties_t props {};
```
Comment (matcabral) [inline scripts/sysman/PROG.rst:385]: props = {}; // missing =

File: scripts/sysman/PROG.rst [hunk 8]
```diff
@@ -424,7 +425,8 @@ sub-device (notice the additional sub-device check):
...
```
Comment (matcabral) [inline scripts/sysman/PROG.rst:428]: props = {}; // missing =

File: scripts/sysman/PROG.rst (hunk 6 above)
Comment (jandres742) [reply scripts/sysman/PROG.rst:799]: @matcabral : initializer `{}` will initialize all members to its default values, including pointers to nullptr.

File: scripts/sysman/PROG.rst (hunk 7 above)
Comment (matcabral) [inline scripts/sysman/PROG.rst:385]: missing =

Comment (matcabral) [reply scripts/sysman/PROG.rst:385]: same for all the occurrences below

**Review by wdamon-intel — APPROVED** [review]
Thanks for patching all those up!

**Pull request merged**